lib/server/nahnojiresolver.py

```diff
@@ -15,21 +15,19 @@ def supports(url):
 
 # returns the steam url
 def url(url):
-	#if supports(url):
 	m = _regex(url)
-	if not m == None:
+	if m:
 		ur = m.group('url')
-		if ur.find('.flv') > -1:
-			return [ur]
-		data = util.substr(util.request(ur),'<body style','</a>')
-		pattern = 'href=\"(.+?)\"[^>]+>' 
-		match = re.compile(pattern).findall(data)
-		return [match[0]]
+		if m.group('flv'): return [ur]
+		page = util.parse_html(ur)
+		data = page.select('source[type=video/mp4]')[0]['src']
+		return ['http://nahnoji.cz'+data]
 
 def resolve(u):
 	stream = url(u)
 	if stream:
 		return [{'name':__name__,'quality':'360p','url':stream[0],'surl':u}]
 
 def _regex(url):
-	return re.search('(?P<url>http://nahnoji.cz/[^\"|\'|\\\]+)',url,re.IGNORECASE | re.DOTALL)
+	return re.search(r'(?P<url>http://nahnoji.cz/((?P<flv>.+\.flv)|(.+)))$',url,re.IGNORECASE | re.DOTALL)
```

Comment:
This overly complex regex is not really necessary and not the most performant one. I'd remove this function and use something like `re.search(r'nahnoji\.cz/.+', url) is not None` in `supports(url)` and `re.search(r'\.flv$', url)` for checking for URLs ending with `.flv`.